pkgs/stdenv/generic/setup.sh

```diff
@@ -429,16 +429,16 @@ findInputs() {
 
         # Host offset relative to the package currently being
         # built---as absolute an offset as will be used.
-        mapOffset relHostOffset
-        local -i hostOffsetNext="$mapOffsetResult"
+        local hostOffsetNext
+        mapOffset "$relHostOffset" hostOffsetNext
 
         # Ensure we're in bounds relative to the package currently
         # being built.
-        [[ "${allPlatOffsets[*]}" = *"$hostOffsetNext"*  ]] || continue
+        (( hostOffsetNext <= 1 && hostOffsetNext >= -1 )) || continue
```

Comment:
Is this equivalent to the original?  I have a hard time seeing how it could be, although I didn't read all the code

Also I usually like to write these conditions like

    (( -1 <= h && h <= 1 ))

So it's like [lower bound, var, var, upper bound]

Reply:
it is equivalent to the original, I thought the string match made it obscur what is actually happening.
I like your style recommendation. Will apply it.

Reply:
I note that the original has a bug, in that it fails to check that `hostOffsetNext` is a *whole word* of `allPlatOffsets`. This probably doesn’t matter for `allPlatOffsets` specifically, although the same bug is repeated later down at `[[ "$visitedPkgs" = *"$pkg"* ]]` where it might matter more.

Reply:
oh, good catch! So the fix would be something like `[[ "$visitedPkgs" = *" $pkgs "* ]]` ?

On another note, do you know why the `[[ -n ${var:-} ]]` is preferred over just `[[ -n $var ]]` ? Or is it just legacy code ?

Reply:
You need the spaces on both sides, in case the match is on one of the ends: `[[ " $visitedPkgs " = *" $pkgs "* ]]`. Although an associative array would likely be a more appropriate data structure there.

`[[ -n "$var" ]]` throws an “unbound variable” error if `set -u` is in effect and `$var` is unset, while `[[ -n "${var:-}" ]]` evaluates to false but continues execution.

Reply:
Thanks a lot!